ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySocketUnitTest.java

```diff
@@ -143,6 +142,35 @@ public void socketScanVerbose() throws Exception {
         }
     }
 
+    @Test
+    public void socketScanTerminateAbruptly() throws Exception {
+        try (Socket connectorSocket = getSocket()) {
+            PrintWriter toServer = new PrintWriter(new OutputStreamWriter(connectorSocket.getOutputStream(), StandardCharsets.UTF_8));
+            toServer.println(RecoveryDriver.SCAN);
+            toServer.flush();
+            recoveryManager.terminate();
+        } catch (final SocketTimeoutException stex) {
+            failOnSocketTimeout(stex, RecoveryDriver.SCAN);
+        }
+    }
+
+    @Test
+    public void socketTimeout() throws Exception {
```

Comment:
Was this expected in here or in JBTM-3257?

Reply:
It's a general test on the socket connection so it could be here but it could be there. It does not hit the specific fixed issue. The socket times out as the input is not finished. This behaviour is the same before the fixes and after the fixes as well.
I can move it if it's preferred.

Reply:
Given your answer, I think the thing to consider is if it was part of the investigation of JBTM-3259 or JBTM-3257 and may ever want backporting or whether it is more applicable to have a separate commit as it is not directly related to either? If an additional commit is necessary I think it could be prefixed it as "ADDITIONAL TEST ONLY:" (like we tend to do for "BUILD SCRIPT ONLY:" and "RELEASE SCRIPT ONLY:") /cc @mmusgrov

Reply:
@tomjenkinson I have really not much an idea what you strive for. I have had the commits separated to three where two were for issues, one was for the maintenance.
Then I was asked to squash them.

Reply:
Hi @ochaloup sorry! I guess I didn't understand exactly what exactly the third commit covered. Let me try to understand the other part of the thread first and I will come back here.